### PR TITLE
Set minimum password length for hotspot

### PR DESCRIPTION
**Pull request by jhbiggs**
#### What has been done to verify that this works as intended?

I've run it through an emulator on API 24 without a problem.

#### Why is this the best possible solution? Were any other approaches considered?

TextWatcher seemed like the best way to  go.  Other solutions involved much more leg work for the same result.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

User will discover that the OK button is greyed out when the password length is less than 8 characters.  A user may stumble on the requirement since there are no explicit instructions.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkCode` and confirmed all checks still pass OR confirm CircleCI build passes

Right, so there was a gradle error, but I can't seem to find the source.  It states that the minimum gradle build should be 4.10.1, but that's already listed in the gradle-wrapper.properties file.  I'm submitting the request anyway as I'm hopeful that a reviewer can identify this issue.  It's not something I changed in my own repository, so I'm puzzled as to why it's an error here.

- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/share/blob/master/share_app/src/main/assets/open_source_licenses.html).

### CODE REVIEW

**Comment by jhbiggs** [comment]
Any suggestions are appreciated to pass the build_debug test.

**Comment by iadeelzafar** [comment]
@jhbiggs  Thanks for making the PR. 
CircleCI is failing and you can find out the reason by having a look at the artifacts.
https://706-133386452-gh.circle-artifacts.com/0/reports/pmd/pmd.html

**Comment by iadeelzafar** [comment]
And you don't need to open a second PR for the same issue. If you're having any issues regarding build, etc. , try finding out a solution and then update that PR with a new commit. If you're unable to find out a solution then I think you should wait for response from the contributors of this project.
Cheers!

**Comment by jhbiggs** [comment]
Ok I've committed the change to my master branch.  The CI builder wanted a final static constant instead of just final.  Not sure how to check it again without creating a new pull request.

**Comment by jhbiggs** [comment]
And how did you get the URL for the nifty CircleCI artifacts page?  All mistakes line by line, very nice.

**Comment by Chromicle** [comment]
> And how did you get the URL for the nifty CircleCI artifacts page? All mistakes line by line, very nice.

if you run `./gradlew checkCode` then you can find the file called `checkstyle.html` as you builds are failing because there is no proper checkstyle
There you can find all your errors or else you can see directly in `CircleCi`

**Comment by Chromicle** [comment]
@jhbiggs why did you changed `gradle-wrapper.properties` it there a particular reason

**Comment by lakshyagupta21** [comment]
@Chromicle its the same he just removed one \ which I think is okay.

**Comment by jhbiggs** [comment]
As I was searching for the CI error I landed on the removing the "\" from the gradle-wrapper.properties file.  It didn't seem to change anything but I will add it back to be safe.

**Comment by jhbiggs** [comment]
Woohoo!  Thanks @Chromicle for the help on checkCode.

File: skunkworks_crow/src/main/java/org/odk/share/views/ui/settings/SettingsActivity.java [hunk 1]
```diff
@@ -142,22 +150,65 @@ private ViewGroup getRootView() {
...
         builder.setCancelable(false);
+        final EditText input = new EditText(this);
+
+        // Specify the type of input expected; this, for example, sets the input as a password,
+        // and will mask the text
+        input.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD);
+        builder.setView(input);
         AlertDialog alertDialog = builder.create();
+
+        //thanks to StackOverflow Aaron for the solution below
```
Comment (huangyz0918) [inline skunkworks_crow/src/main/java/org/odk/share/views/ui/settings/SettingsActivity.java]: If you really want to thank that, can you provide a link here?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -4,3 +4,5 @@ distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
 distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.1-all.zip
+
```
Comment (huangyz0918) [inline gradle/wrapper/gradle-wrapper.properties]: I think you can remove the two-lines change since it's unnecessary.

**Comment by lakshyagupta21** [comment]
@jhbiggs Are you still working on the comments suggested by @huangyz0918 ?

**Comment by Chromicle** [comment]
@jhbiggs your build is failing because of checkstyle to make build pass you can run `./gradlew checkcode` you can get the detail information why your build is failing

**Comment by jhbiggs** [comment]
I removed some lines from the Settings activity per the CircleCI recommendation.  Will see what turns up and take the next step.

**Comment by lakshyagupta21** [comment]
@jhbiggs Sorry for keeping this PR in review state for long,  are you still working on this as I can see the tests are still failing in the PR as well there are some merge conflicts. 
And please update the PR description and mention about the github issue you're working on so that if someone else goes through the issue they can refer to the PR and get the idea that someone has already raised the PR. 
Just for reference this PR is raised for #233

**Comment by jhbiggs** [comment]
Hi @lakshyagupta21 I'm confused here.  I saw the fail reason for the build_debug test, but it had nothing to do with my changes.  Do you have a suggestion?  I don't want to chase my tail here unnecessarily if there was a recent change that might solve this problem.

**Comment by manit-alokbatham** [comment]
@opendatakit-bot claim

**Comment by lakshyagupta21** [comment]
@manit-alokbatham Please claim issues only on which PRs are not raised.

**Comment by lakshyagupta21** [comment]
Closing this PR as @jhbiggs agreed to let @devanshi7799 work on this. 

Thanks @jhbiggs for giving new contributors a chance to contribute to open source.

**Pull request closed without merging**